packages/http/src/router/matchPath.ts

```diff
@@ -1,18 +1,29 @@
 import { MatchType } from './types';
 import * as E from 'fp-ts/Either';
 
-function fragmentarize(path: string): string[] {
+function fragmentize(path: string): string[] {
   return path.split('/').slice(1);
 }
 
-function getTemplateParamName(pathFragment: string) {
-  const match = /{(.*)}/.exec(pathFragment);
+// Attempt to decode path fragment. Decode should not do any harm since it is
+// decoding only URI sequences which are previously created by encodeURIComponent
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent
+function decodePathFragment(pathFragment?: string) {
+  try {
+    return pathFragment && decodeURIComponent(pathFragment);
+  } catch (_) {
+    return pathFragment;
```

Comment:
when `decodeURIComponent` can throw an error?

Reply:
It'll throw if you provide any invalid sequence, for example: `decodeURIComponent('%1')`. I just want to make sure that we can handle some weird URLs.